app/config/AWS.scala

```diff
@@ -3,52 +3,69 @@ package config
+import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import com.amazonaws.services.simpleemail.AmazonSimpleEmailServiceClientBuilder
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClientBuilder
 import com.amazonaws.util.EC2MetadataUtils
 
 import scala.collection.JavaConverters._
 
+case class InstanceTags(stack: String, app: String, stage: String)
 
 object AWS {
-  lazy val region = Region getRegion Regions.EU_WEST_1
+  lazy val region = Region.getRegion(Regions.EU_WEST_1).getName
   lazy val workflowAwsCredentialsProvider = new AWSCredentialsProviderChain(
     new EnvironmentVariableCredentialsProvider(),
     new SystemPropertiesCredentialsProvider(),
-    new InstanceProfileCredentialsProvider(),
+    InstanceProfileCredentialsProvider.getInstance(),
     new ProfileCredentialsProvider("workflow")
   )
   lazy val composerAwsCredentialsProvider = new AWSCredentialsProviderChain(
     new EnvironmentVariableCredentialsProvider(),
     new SystemPropertiesCredentialsProvider(),
-    new InstanceProfileCredentialsProvider(),
+    InstanceProfileCredentialsProvider.getInstance(),
     new ProfileCredentialsProvider("composer")
   )
 
-  def eC2Client = region.createClient(classOf[AmazonEC2Client], workflowAwsCredentialsProvider, null)
-  def s3Client = region.createClient(classOf[AmazonS3Client], composerAwsCredentialsProvider, null)
-  lazy val sesClient = region.createClient(classOf[AmazonSimpleEmailServiceClient],
-    composerAwsCredentialsProvider, null)
-
-  def dynamoDbClient = region.createClient(classOf[AmazonDynamoDBClient], composerAwsCredentialsProvider, null)
-
-  def getInstanceId = Option(EC2MetadataUtils.getInstanceId)
-
-  def readTag(tagName: String, instanceId: Option[String])(implicit ec2Client: AmazonEC2Client): Option[String] = {
-    instanceId.flatMap { id =>
-      val tagsResult = ec2Client.describeTags(
-        new DescribeTagsRequest().withFilters(
-          new Filter("resource-type").withValues("instance"),
-          new Filter("resource-id").withValues(id),
-          new Filter("key").withValues(tagName)
-        )
-      )
-      tagsResult.getTags.asScala
-        .find(_.getKey == tagName)
-        .map(_.getValue)
+  lazy val asgClient = AmazonAutoScalingClientBuilder.standard().withRegion(region).withCredentials(composerAwsCredentialsProvider).build()
+  lazy val ssmClient = AWSSimpleSystemsManagementClientBuilder.standard().withRegion(region).withCredentials(composerAwsCredentialsProvider).build()
+
+  lazy val s3Client = AmazonS3ClientBuilder.standard().withRegion(region).withCredentials(composerAwsCredentialsProvider).build()
+  lazy val sesClient = AmazonSimpleEmailServiceClientBuilder.standard().withRegion(region).withCredentials(composerAwsCredentialsProvider).build()
+  lazy val dynamoDbClient = AmazonDynamoDBClientBuilder.standard().withRegion(region).withCredentials(composerAwsCredentialsProvider).build()
+
+  def readTags(): Option[InstanceTags] = {
```

Comment:
@aug24 here's the code that reads the tags from the autoscaling group.

The IAM permissions in cloudformation are in a different repo to this one but it worked because we already had `ec2:Describe*`. I think the actual permissions required are `ec2:DescribeAutoScalingGroups` and `ec2:DescribeAutoScalingInstances`.